vTPM Support in Cloud-Hypervisor (#4642)
cloud-hypervisor/cloud-hypervisor: docs/vtpm.md

> # vTPM

Reviewer: I would really try to avoid the `v` in front of TPM given it's implicitly always virtual. The VMM emulates a TPM device, which is why it can be seen as virtual.

> vtpm in Cloud-Hypervisor is emulated while using ```swtpm``` in the backend.

Reviewer: ```suggestion
`tpm` in Cloud-Hypervisor is emulated using `swtpm` as the backend.
```

Current implementation only supports TPM ```2.0``` version. At the moment only
```CRB Interface``` is implementation. This interface is described in
[TCG PC Client Platform TPM Profile Specification for TPM 2.0, Revision 01.05 v4](https://trustedcomputinggroup.org/wp-content/uploads/PC-Client-Specific-Platform-TPM-Profile-for-TPM-2p0-v1p05p_r14_pub.pdf).

## Usage

> `--vtpm-socket`, an optional argument, can be passed to enable vtpm device.

Reviewer: I would prefer a more generic option here. A simple `--tpm` is what I would expect, which would accept `socket=` as a parameter. And if someone wants to plug a different type of emulated TPM in the future, we might add a different parameter.

This argument takes an UNIX domain Socket as a value.

_Example_

An Example invocation with --vtpm-socket argument:

```
 ./cloud-hypervisor/target/release/cloud-hypervisor \
	--kernel ./hypervisor-fw \
	--disk path=focal-server-cloudimg-amd64.raw \
	--cpus boot=4 \
	--memory size=1024M \
	--net "tap=,mac=,ip=,mask=" \

>     --vtpm-socket="/var/run/swtpm.socket"

Reviewer 2: For consistency this should be `--vtpm socket=`

Reviewer: I think this should be `--tpm` instead of `--vtpm`.
